internal/server/v1beta1/search_test.go

```diff
@@ -332,3 +335,170 @@ func TestSuggest(t *testing.T) {
+					},
+				}
+				as.EXPECT().GroupAssets(ctx, cfg).Return([]asset.GroupResult{}, nil)
+			},
+		},
+		{
+			Description: "should pass include fields to search config format",
+			Request: &compassv1beta1.GroupAssetsRequest{
+				Groupby: []string{"resource"},
+				Filter: map[string]string{
+					"data.landscape": "th",
+					"type":           "topic",
+					"service":        "kafka,rabbitmq",
+				},
+				IncludeFields: []string{"data.columns.name", "owners.email"},
+			},
+			Setup: func(ctx context.Context, as *mocks.AssetService) {
+				cfg := asset.GroupConfig{
+					GroupBy: []string{"resource"},
+					Filters: map[string][]string{
+						"type":           {"topic"},
+						"service":        {"kafka", "rabbitmq"},
+						"data.landscape": {"th"},
+					},
+					IncludedFields: []string{"data.columns.name", "owners.email"},
+				}
+				as.EXPECT().GroupAssets(ctx, cfg).Return([]asset.GroupResult{}, nil)
+			},
+		},
+		{
+			Description: "should return the grouped documents",
+			Request: &compassv1beta1.GroupAssetsRequest{
+				Groupby: []string{"resource"},
+			},
+			Setup: func(ctx context.Context, as *mocks.AssetService) {
+				cfg := asset.GroupConfig{
+					GroupBy: []string{"resource"},
+					Filters: make(map[string][]string),
+				}
+				response := []asset.GroupResult{
+					{
+						Fields: []asset.GroupField{{
+							Name:  "resource",
+							Value: "kafka",
+						},
+						},
+						Assets: []asset.Asset{{
+							Type:        "test",
+							ID:          "test-resource",
+							Description: "some description",
+							Service:     "test-service",
+							Labels: map[string]string{
+								"entity":    "gotocompany",
+								"landscape": "id",
+							},
+						},
+						},
+					},
+				}
+				as.EXPECT().GroupAssets(ctx, cfg).Return(response, nil)
+			},
+
+			PostCheck: func(resp *compassv1beta1.GroupAssetsResponse) {
+				expected := &compassv1beta1.GroupAssetsResponse{
+					AssetGroups: []*compassv1beta1.AssetGroup{
+						{
+							GroupFields: []*compassv1beta1.GroupField{
+								{
+									GroupKey:   "resource",
+									GroupValue: "kafka",
+								},
+							},
+							Assets: []*compassv1beta1.Asset{
+								{
+									Id:          "test-resource",
+									Description: "some description",
+									Service:     "test-service",
+									Type:        "test",
+									Labels: map[string]string{
+										"entity":    "gotocompany",
+										"landscape": "id",
+									},
+								},
+							},
+						},
+					},
+				}
+				testutils.AssertEqualProto(t, expected, resp)
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.Description, func(t *testing.T) {
+			ctx := user.NewContext(context.Background(), user.User{UUID: userUUID})
+
+			logger := log.NewNoop()
+			mockUserSvc := new(mocks.UserService)
+			mockSvc := mocks.NewAssetService(t)
+			if tc.Setup != nil {
+				tc.Setup(ctx, mockSvc)
+			}
+
+			defer mockUserSvc.AssertExpectations(t)
+			defer mockSvc.AssertExpectations(t)
```

Comment:
```suggestion
			mockUserSvc := mocks.NewUserService(t)
			mockSvc := mocks.NewAssetService(t)
			if tc.Setup != nil {
				tc.Setup(ctx, mockSvc)
			}
```

please take some time to go through the review comment and address it thoroughly. Would appreciate it if multiple follow-ups are not required for addressing an issue 🙏🏼

Reply:
I mistakenly thought it was about the AssetService instead of user service, for now I have changed for both. Resolving this conversation for now.